design/codeset.go

```diff
@@ -0,0 +1,141 @@
+package design
+
+import (
+	. "goa.design/goa/v3/dsl"
+)
+
+var _ = Service("codeset", func() {
+	Description("The codeset service performs operations on Codesets.")
+
+	// Method describes a service method (endpoint)
+	Method("list", func() {
+		Description("Retrieve information about Codesets registered in FuseML.")
+		// Payload describes the method payload.
+		// Here the payload is an object that consists of two fields.
+		Payload(func() {
+			// Field describes an object field given a field index, a field
+			// name, a type and a description.
+			Field(1, "project", String, "List only Codesets that belong to given project", func() {
+				Example("mlflow-project-01")
+			})
+			Field(2, "label", String, "List only Codesets with matching label", func() {
+				Example("mlflow")
+			})
+		})
+
+		// Result describes the method result.
+		// Here the result is a collection of codeset value.
+		Result(ArrayOf(Codeset), "Return all registered Codesets matching the query.")
+
+		Error("NotFound", func() {
+			Description("If the Codeset is not found, should return 404 Not Found.")
+		})
+
+		// HTTP describes the HTTP transport mapping.
+		HTTP(func() {
+			// Requests to the service consist of HTTP GET requests.
+			// The payload fields are encoded as path parameters.
+			GET("/codesets")
```

Comment:
We should use the same convention regarding naming for REST API paths, either all plural (runnables, codesets), or all singular.

Reply:
I think I have plural everywhere, just like with runnables. Or do you think it should be `GET("/codeset")` when retrieving single item?

Reply:
Nevermind, for some reason I thought the runnable goa design was using `runnable` instead of `runnables`. I stand corrected.